analytics/account_creation.js

```diff
@@ -2,10 +2,48 @@
 // Generic events for tracking events around account creation.
 //
 
-// Created account
-analyticsHooks.on('auth:register', function (options) {
-  analytics.track('Created account')
-})
+// Created account (via email)
+$(document).on(
+  'submit',
+  '.auth-register form, .marketing-signup-modal form',
+  function() {
+    $(document).one('ajaxComplete', function(e, xhr) {
+      analytics.track('Created account', {
+        acquisition_initiative: location.search.replace('?m-id=', ''),
+        signup_service: 'email',
+        user_id: xhr.responseJSON.user.id
+      })
+    })
+  }
+)
+
+// Created account (via social)
+
+// 1. Upon clicking the social signup button
+$(document).on(
+  'click',
+  '.auth-signup-facebook, .marketing-signup-modal-fb',
+  function() {
+
+    // 2. Store some data in cookies before being redirected everywhere
+    Cookies.set('analytics-signup', JSON.stringify({
+      service: 'facebook',
+      acquisition_initiative: location.search.replace('?m-id=', '')
+    }))
+  }
+)
+
+// 3. After landing back on Artsy send the tracking call and expire the cookie
+if (Cookies.get('analytics-signup')) {
+  var data = JSON.parse(Cookies.get('analytics-signup'))
```

Comment:
This explains the hoops we're jumping through to get our sophisticated signup spying working. Just like we need to persist some state in the session to follow the user around the web [on the server-side](https://github.com/artsy/artsy-passport/pull/88#discussion_r90137532), we do roughly the same thing here on the client-side using Cookies.js to remember the Facebook button click, then when the come back to Artsy after successful OAuth signup we make the tracking call here.